beetsplug/lyrics.py

```diff
@@ -493,14 +494,31 @@ def parse_search_results(self, html):
 
         return self.BASE_URL + link.get('href')
 
-    def extract_lyrics(self, html):
+    def extract_lyrics(self, html, artist, title):
         html = _scrape_strip_cruft(html)
         html = _scrape_merge_paragraphs(html)
 
         soup = try_parse_html(html)
         if not soup:
             return None
 
+        info_div = soup.find("div", class_="col-auto")
+        if not info_div:
+            return None
+
+        info_elements = info_div.find_all("a")
+        if not info_elements:
+            return None
+
+        html_title = info_elements[-1].get_text()
+        html_artist = info_elements[-2].get_text()
+
+        title_dist = string_dist(html_title, title)
+        artist_dist = string_dist(html_artist, artist)
+
+        if title_dist > 0.1 or artist_dist > 0.1:
```

Comment:
What do you think about making this configurable? Even if we make it completely undocumented (probably a good idea; not much of a reason to customize this), separating out the magic number will make it easy to adapt this in the future if we need to.

This would require something like:


```py
thresh = self.config['dist_thresh'].get(float)
if title_dist > thresh or artist_dist > thresh:
```

…as well as adding the appropriate key to the `self.config.add(...)` call in `LyricsPlugin` to set the default.

Reply:
Yes, that would make sense! I just committed a change that adds this configurable distance threshold.